### DISCUSSION TITLE
Rollow ZMK support

### DISCUSSION THREAD

**Discussion by Joeysface** [manna-harbour/miryoku#107] [Q&A] [answered]
Hi, I was wondering if there were plans to officially support rollow in the miryoku zmk repo?

**Comment by manna-harbour** [comment, marked as answer]
I'm happy to add support for new keyboards by request, see https://github.com/manna-harbour/miryoku/discussions/81.

I've added rollow now in https://github.com/manna-harbour/miryoku_zmk/commit/c54acd2523d9da86a20a7f4f3bb8c93ccdd95cc7. Encoder press acts as the tertiary thumb key, but thumb combos are also enabled, see https://github.com/manna-harbour/miryoku_zmk#thumb-combos.

However the shield needs updating for the current version of ZMK, see https://zmk.dev/blog/2022/04/02/zephyr-3-0#rgb-underglow. Until then you'll have to build with an older version, see https://github.com/manna-harbour/miryoku/discussions/81. Here's a sample build run for current and older ZMK: https://github.com/manna-harbour/miryoku_zmk/actions/runs/2661925045

**Reply by manna-harbour** [reply]
I'll add proper support for encoder sensor bindings to Miryoku ZMK soon, but here's something you can use for the time being: https://github.com/manna-harbour/miryoku_zmk/commit/7430ce30ba4ce14004f41a6f05d8cfdf23af7fa8.

To enable peripheral side encoder support in ZMK you can use `infused-kim/zmk/sofle` in the workflow `branches` option, see https://github.com/manna-harbour/miryoku_zmk#branches.

Here's a test build using both of those: https://github.com/manna-harbour/miryoku_zmk/actions/runs/2681072436.

**Reply by evernal** [reply]
Both of these test build outputs expired (and the logs that I could crib from on your build inputs), so I tried to reproduce in my ZMK fork: https://github.com/evernal/miryoku_zmk/actions/runs/3357206687 but my encoders don't seem to be working. Do I need to set a custom configuration in build inputs to take advantage of the 7430ce3 change above? Thanks!

**Reply by manna-harbour** [reply]
> Both of these test build outputs expired (and the logs that I could crib from on your build inputs), so I tried to reproduce in my ZMK fork: https://github.com/evernal/miryoku_zmk/actions/runs/3357206687 but my encoders don't seem to be working. Do I need to set a custom configuration in build inputs to take advantage of the 7430ce3 change above? Thanks!

To map encoder rotations, the commit that was added to the `encoder-test` branch needs to be in the branch you're building from, e.g. by [adding that branch to your fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository) and selecting it when running the workflow. (In general, it'd be better to cherry pick or merge into a new branch based on master, but I just rebased `encoder-test`, so at the moment it's up to date.) Then to customise the encoder mappings you could do it in `custom_config.h` by substituting a custom `MIRYOKU_LAYER_LIST` based on the one from that commit (https://github.com/manna-harbour/miryoku_zmk/blob/22bd99f5af88d2bc38a12503b5cd1e7eaab5c088/miryoku/miryoku_babel/miryoku_layer_list.h#L9-L19).

Here's a new build run from the rebased `encoder-test` branch for `nice_nano_v2`, `rollow_left,rollow_right`, and `infused-kim/zmk/sofle`: https://github.com/manna-harbour/miryoku_zmk/actions/runs/3382996117.

**Reply by evernal** [reply]
Thank you! I just tried the build you did and the encoders work as expected. I'd like to make sure to "learn to fish" here too and will go validate that I can do my own build too.
